src/AutoMapper/PropertyMap.cs

```diff
@@ -38,7 +39,7 @@ public PropertyMap(PropertyMap inheritedMappedProperty, TypeMap typeMap)
         public LambdaExpression CustomResolver { get; set; }
         public LambdaExpression Condition { get; set; }
         public LambdaExpression PreCondition { get; set; }
-        public LambdaExpression CustomExpression { get; private set; }
+        public LambdaExpression CustomExpression { get; internal set; }
```

Comment:
Let's see if we can make this a private property and maybe an interface/base class so that the plan builder can use both a PropertyMap and a PathMap.